src/raphael_backend_flask/multimodal.py

```diff
@@ -0,0 +1,91 @@
+import json
+import re
+import os
+import tempfile
+
+from google.cloud import storage
+import yt_dlp
+
+from raphael_backend_flask.db import create_multimodal_claim_extraction_run
+from raphael_backend_flask.exceptions import FlashException
+
+GCS_BUCKET = "fullfact-nlp"
+GCS_FOLDER = "raphael/videos"
+
+# Not used at the moment as we're not restricting multimodal to YouTube
+tiktok_url_matcher = re.compile(r"https?://(www\.)?tiktok.com/[^/]+/video/([0-9]+).*")
+
+
+def handle_multimodal_url(user_id: int, url: str) -> int:
+    info = fetch_video(url)
+    if not info:
+        raise FlashException("Could not process video.")
+
+    title = info.get("title")
+    if not title or title.startswith("Video by"):
+        desc = info.get("description")
+        if desc and len(desc) < 150:
+            title = desc
+
+    metadata = {
+        "title": title,
+        "url": info["webpage_url"],
+    }
+
+    return create_multimodal_claim_extraction_run(
+        user_id, info["dest"], json.dumps(metadata)
+    )
+
+
+def upload_file_to_gcs(local_filepath: str, bucket_name: str, folder: str) -> str:
+    filename = local_filepath.split("/")[-1]
+    storage_client = storage.Client()
```

Comment:
Clients are thread safe and so you could initialise this at the module level instead of per-call.